crates/interpreter/src/parser.rs

```diff
@@ -272,6 +284,30 @@ impl<'m, M: Mode> Parser<'m, M> {
         Ok(Instr::BrTable(labels, self.parse_labelidx()?))
     }
 
+    #[cfg(feature = "threads")]
+    pub fn parse_instr_fe(&mut self) -> MResult<Instr<'m>, M> {
+        Ok(match self.parse_byte()? {
+            0 => Instr::AtomicNotify(self.parse_memarg()?),
+            x @ 1 ..= 2 => Instr::AtomicWait((x - 1).into(), self.parse_memarg()?),
+            3 => {
+                check_eq::<M, _>(self.parse_byte()?, 0)?;
+                Instr::AtomicFence
+            }
+            x @ 16 ..= 17 => Instr::AtomicLoad((x - 16).into(), self.parse_memarg()?),
+            x @ 18 ..= 22 => Instr::AtomicLoad_((x - 18).into(), self.parse_memarg()?),
+            x @ 23 ..= 24 => Instr::AtomicStore((x - 23).into(), self.parse_memarg()?),
+            x @ 25 ..= 29 => Instr::AtomicStore_((x - 25).into(), self.parse_memarg()?),
+            x @ 30 ..= 78 => {
+                let op: AtomicOp = ((x - 30) / 7).into();
+                match (x - 30) % 7 {
+                    x @ 0 ..= 1 => Instr::AtomicOp(x.into(), op, self.parse_memarg()?),
+                    x => Instr::AtomicOp_((x - 2).into(), op, self.parse_memarg()?),
+                }
+            }
+            _x => M::unsupported(if_debug!(Unsupported::Opcode(_x)))?,
```

Comment:
That's similar to the top-level byte. You can add your own debug-print for now with a TODO(threads), but those bytes are always invalid for the spec we target.
```suggestion
            _ => M::invalid()?,
```